### PR TITLE
Use Option for CallContract member 'data'

### PR DESCRIPTION
**Pull request by tcharding**
Currently we use `bytes.default()` to generate an empty value for byte array `data` member of the `CallContract` structure.  This results in the value `0x` being used (I do not know the exact reason why or where this `0x` is found, please see note below on testing).  Rust has options for such a case.

This needs testing, I have no idea what will now be expressed by the `None` in place of the original `0x`.

Fixes: #999

### CODE REVIEW

**Comment by thomaseizinger** [comment]
`CallContract` gets converted to `ActionResponseBody::EthereumCallContract`. 

What do you think @D4nte and @luckysori in terms of the API?
I think we want a test that the `data` field just does not exist if we serialize `ActionResponseBody::EthereumCallContract` to JSON and the `Option` is `None`.

**Comment by D4nte** [comment]
> `CallContract` gets converted to `ActionResponseBody::EthereumCallContract`.
> 
> What do you think @D4nte and @luckysori in terms of the API?
> I think we want a test that the `data` field just does not exist if we serialize `ActionResponseBody::EthereumCallContract` to JSON and the `Option` is `None`.

Yes a serialisation test when `data` is `None` to confirm it's not present in the JSON would be great please.

**Comment by thomaseizinger** [comment]
> > `CallContract` gets converted to `ActionResponseBody::EthereumCallContract`.
> > What do you think @D4nte and @luckysori in terms of the API?
> > I think we want a test that the `data` field just does not exist if we serialize `ActionResponseBody::EthereumCallContract` to JSON and the `Option` is `None`.
> 
> Yes a serialisation test when `data` is `None` to confirm it's not present in the JSON would be great please.

@tcharding Go for that then :)

**Comment by tcharding** [comment]
The enum variant `ActionResponseBody::EthereumCallContract` uses a struct that has fields that mirror the `ethereum::CallContract` struct.  This is duplication of knowledge, we can just use the `ethereum::CallContract` directly in the enum variant instead of a struct.  This allows us to rely on the serialization testing done for `CallContract`.  The bitcoin variants are probably able to be done the same way but its Friday afternoon so this is all for this PR :)  Can be merged without doing the bitcoin variants IMO.

**Comment by D4nte** [comment]
Taking over, I will just remove commit 601aa41391d7360b3cc4a2a54593f9af64353df0

File: comit_node/src/swap_protocols/actions.rs [hunk 1]
```diff
@@ -52,12 +55,57 @@ pub mod ethereum {
...
     pub struct CallContract {
         pub to: Address,
-        pub data: Bytes,
+        pub data: Option<Bytes>,
         pub gas_limit: U256,
         pub network: Network,
         pub min_block_timestamp: Option<Timestamp>,
     }
+
+    impl Serialize for CallContract {
```
Comment (thomaseizinger) [inline comit_node/src/swap_protocols/actions.rs]: This implementation should not exist.

File: comit_node/src/swap_protocols/actions.rs [hunk 2]
```diff
@@ -52,12 +55,57 @@ pub mod ethereum {
...
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use ethereum_support::{web3::types::U256, Address, Network};
+    use std::str::FromStr;
+
+    #[test]
+    fn call_contract_serializes_correctly_to_json_with_none() {
```
Comment (thomaseizinger) [inline comit_node/src/swap_protocols/actions.rs]: This test should be testing what is now actually serialized to JSON, which is `ActionResponsePayload::EthereumCallContract`.

**Comment by D4nte** [comment]
Looks like I messed up the commits by rebasing master after Tobin merged master. Will do fresh PR shortly.

File: api_tests/lib/ethereum.ts [hunk 3]
```diff
@@ -147,7 +147,7 @@ export class EthereumWallet {
 
     public async sendEthTransactionTo(
         to: string,
-        data: string = "0x0",
```
Comment (thomaseizinger) [inline api_tests/lib/ethereum.ts]: This stupid `EthereumWallet` has bitten us so many times now, on `bobtimus` as well as here in the tests.
I think we should extract what we have here into an NPM package and publish it. Basically an idiot-proof `EthereumWallet` that cannot be used wrongly :D

Comment (D4nte) [reply api_tests/lib/ethereum.ts]: hahahahaha, yes I think it's worth considering and to give back to the community.

Comment (thomaseizinger) [reply api_tests/lib/ethereum.ts]: Created ticket here: https://github.com/coblox/bobtimus/issues/78
